### PR TITLE
CAT-2250 Don't display a think/say bubble when the text equals ""

### PR DESCRIPTION
**Pull request by rschluesselbauer**
Added a check in ThinkSayBubbleAction:
When the text equals "", don't create a new bubble actor and don't add it to the stage.

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/content/actions/ThinkSayBubbleAction.java [hunk 1]
```diff
@@ -47,14 +47,15 @@ protected void update(float delta) {
 			Log.d(getClass().getSimpleName(), "Formula interpretation for this specific Brick failed.", interpretationException);
 			return;
 		}
-		ShowBubbleActor showBubbleActor = new ShowBubbleActor(textToDisplay, sprite, type);
-
 		if (StageActivity.stageListener.getBubbleActorForSprite(sprite) != null) {
```
Comment (loibi93) [inline catroid/src/main/java/org/catrobat/catroid/content/actions/ThinkSayBubbleAction.java]: If you have a program:

ShowBubble "Hello"
Wait 10 Seconds
ShowBubble ""

The Bubble with "Hello" would disappear, is this intended ?

Comment (robertpainsi) [inline catroid/src/main/java/org/catrobat/catroid/content/actions/ThinkSayBubbleAction.java]: image
According to Scratch, yes

**Comment by robertpainsi** [comment]
ok to test

**Comment by rschluesselbauer** [comment]
There has been an addition to the requirements so this PR is on hold ATM.
A formula element that is completely empty should get the value of an empty string, if the brick expects a string.

To the people who know their way around Jenkins - I am very confused about the results of the build. Could you explain to me why I get checkstyle errors for files I didn't even touch?

**Comment by robertpainsi** [comment]
@rschluesselbauer, about Checkstyle, don't worry about it :wink: Jenkins tests the merge of both branches, in this case `Catrobat:develop` and `rschluesselbauer:CAT-2250`, and there [have been Checkstyle warnings](https://github.com/Catrobat/Catroid/pull/2261) in the develop branch.

You already did the right thing :1st_place_medal:! Instead of fixing and adding unrelated code to this PR, which is a really bad thing to do, you mentioned that there is a problem.

About this PR, please also add a test for this behavior. If possible, a unit test.

**Comment by robertpainsi** [comment]
> A formula element that is completely empty…

How can a formula element be completely empty or what do you mean by that?

**Comment by rschluesselbauer** [comment]
By editing the speak bubble text and removing everything. Normally the resulting formula is not accepted, but according to the new requirements it should be accepted for specific blocks (like thinkbubble or speak or note) and replaced by a formula that represents an empty string.

**Comment by robertpainsi** [comment]
@rschluesselbauer, ah, now I get it. Thanks for the update :smile_cat:

**Pull request closed without merging**